crates/holochain/src/core/ribosome/host_fn.rs

```diff
@@ -23,14 +23,14 @@ macro_rules! host_fn_api_impls {
         )*
 
         impl<Ribosome: RibosomeT> HostFnApiT for HostFnApi<Ribosome> {
-            type Error = RibosomeError;
             $(
-                fn $f(&self, input: $input) -> Result<$output, Self::Error> {
-                    $f::$f(
+                fn $f(&self, input: $input) -> Result<$output, HostFnApiErrorPlaceholder> {
+                    let out = $f::$f(
                         self.ribosome.clone(),
                         self.call_context.clone(),
                         input.into()
-                    ).map(|r| r.into_inner())
+                    ).map(|r| r.into_inner());
+                    Ok(out.expect("TODO, handle after crate re-org [B-03640]"))
```

Comment:
HostFnApi is only being used in InlineZome currently, so only InlineZome. Maybe we will use HostFnApi everywhere at some point, but we should do the crate refactor before then anyway, so *hopefully* this will get resolved soon

Reply:
The reason for this is that the HostFnApiT trait can't know about RibosomeError since it's downstream. Hence the motivation to colocate Zome and Ribosome